domains/oidc-mock/Mock/Controllers/AuthController.cs

```diff
@@ -0,0 +1,106 @@
+using Oidc.Mock.Extensions;
+using Oidc.Mock.Jwt;
+using Oidc.Mock.Models;
+using System.Net.Http.Headers;
+
+namespace Oidc.Mock.Controllers;
+
+public class AuthController : Controller
+{
+    private readonly Client _client;
+    private readonly User[] _users;
+    private readonly IJwtTokenGenerator _tokenGenerator;
+    private readonly ILogger<AuthController> _logger;
+
+    public AuthController(Client client, User[] users, IJwtTokenGenerator tokenGenerator, ILogger<AuthController> logger)
+    {
+        _client = client;
+        _users = users;
+        _tokenGenerator = tokenGenerator;
+        _logger = logger;
+    }
+
+    [HttpPost]
+    [Route("api/v1/session/logout")]
+    public IActionResult LogOut() => Ok();
+
+    [HttpGet]
+    [Route("Connect/Authorize")]
+    public IActionResult Authorize(string client_id, string redirect_uri)
+    {
+        var (isValid, validationError) = _client.Validate(client_id, redirect_uri);
+        if (!isValid)
+        {
+            return BadRequest(validationError);
+        }
+
+        var routeValues = new RouteValueDictionary();
+        foreach (var keyValuePair in Request.Query)
+        {
+            routeValues.Add(keyValuePair.Key, keyValuePair.Value);
+        }
+
+        return RedirectToPage("/Connect/Signin", routeValues);
+    }
+
+    [HttpPost]
+    [Route("Connect/Token")]
+    public IActionResult Token(string grant_type, string code, string redirect_uri)
+    {
+        var authorizationHeader = AuthenticationHeaderValue.Parse(Request.Headers["Authorization"]);
+        _logger.LogDebug($"connect/token: authorization header: {authorizationHeader.Scheme} {authorizationHeader.Parameter}");
+        _logger.LogDebug($"connect/token: form data: {string.Join("; ", Request.Form.Select(kvp => $"{kvp.Key}={kvp.Value}"))}");
+
+        if (!string.Equals(grant_type, "authorization_code", StringComparison.InvariantCultureIgnoreCase))
+        {
+            return BadRequest($"Invalid grant_type. Must be 'authorization_code', but was '{grant_type}'");
+        }
+
+        var auth = (authorizationHeader.Parameter ?? ":").DecodeBase64();
+        var split = auth.Split(":");
+        var clientId = split[0];
+        var clientSecret = split[1];
+
+        var (isValid, validationError) = _client.Validate(clientId, clientSecret, redirect_uri);
+        if (!isValid)
+        {
+            _logger.LogDebug($"connect/token: {validationError}");
+            return BadRequest(validationError);
+        }
+
+        var user = _users.FirstOrDefault(u => string.Equals(u.Name?.ToMd5(), code));
+        if (user == null)
+        {
+            _logger.LogDebug("connect/token: Invalid code - no matching user");
+            return BadRequest("Invalid code - no matching user");
+        }
+
+        var now = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+        var baseClaims = new Dictionary<string, object>
+        {
+            { "iss", $"{Request.Scheme}://{Request.Host}" },
```

Comment:
Note to reviewer: This will not result in the correct uri for issuer as the scheme when running behind Traefik will be "http" and not "https". Furthermore, PATH_BASE env. variable should be appended. But it is not an issue with current auth service, so I have left this as is.